File: libs/eave-stdlib-ts/src/core-api/enums.ts
export enum AuthProvider {
  google = 'google',
  slack = 'slack',
  atlassian = 'atlassian',
}

export enum DocumentPlatform {
  eave = 'eave',
  confluence = 'confluence',
  google_drive = 'google_drive',
}

export enum SubscriptionSourcePlatform {
  slack = 'slack',
  github = 'github',
  jira = 'jira',
}

export enum SubscriptionSourceEvent {
  slack_message = 'slack_message',
  github_file_change = 'github_file_change',
  jira_issue_comment = 'jira_issue_comment',
}

/**
 * Link types that we support fetching content from for integration into AI documentation creation.
 */
export enum LinkType {
  github = 'github',
}

export enum EaveForgeInboundOperation {
  createDocument = 'createDocument',
  updateDocument = 'updateDocument',
  archiveDocument = 'archiveDocument',

}

File: apps/core/eave/core/enums.py
from enum import StrEnum
import strawberry.federation as sb

@sb.enum
class DocumentPlatform(StrEnum):
    eave = "eave"
    confluence = "confluence"
    google_drive = "google_drive"

@sb.enum
class AuthProvider(StrEnum):
    google = "google"
    slack = "slack"
    atlassian = "atlassian"
    github = "github"


@sb.enum
class LastJobResult(StrEnum):
    none = "none"
    doc_created = "doc_created"
    no_api_found = "no_api_found"
    error = "error"


@sb.enum
class ApiDocumentationJobState(StrEnum):
    running = "running"
    idle = "idle"

@sb.enum
class Integration(StrEnum):
    """
    Apps that a Team can integrate with.
    """

    slack = "slack"
    github = "github"
    atlassian = "atlassian"
    confluence = "confluence"
    jira = "jira"

@sb.enum
class AtlassianProduct(StrEnum):
    jira = "jira"
    confluence = "confluence"
